src/api/github.js

```diff
@@ -21,6 +24,39 @@ export const getUserByPat = async token => {
 
 export const getEntity = async (entityInfo, token) => {
   const { owner, repo, type, number } = entityInfo
+
+  if (type === 'discussions') {
+    try {
+      const result  = await getGraphqlClient()
+      .query(DISCUSSION_QUERY, {
+        number: parseInt(number),
+        owner: owner,
+        name: repo
+      })
+      .toPromise()
+      .then(result => {
+        if (result.error) {
+          return {
+            error: 'Error while fetching the discussion. Fine-grained PATs are not supported yet.'
+          }
+        }
+
+        return result.data
+      });
+
+      if (result.error) {
+        return result;
+      }
+      return {
+        updated_at: result?.repository.discussion.updatedAt
+      }
+    } catch (error) {
+      return {
+        error
+      }
+    }
```

Comment:
refactor into a function please for readability and leverage async await

Reply:
Function extracted.